src/test/groovy/unit/org/pih/warehouse/receiving/ReceiptServiceSpec.groovy

```diff
@@ -0,0 +1,105 @@
+package unit.org.pih.warehouse.receiving
+
+import grails.testing.gorm.DataTest
+import grails.testing.services.ServiceUnitTest
+import org.pih.warehouse.api.PartialReceipt
+import org.pih.warehouse.core.ActivityCode
+import org.pih.warehouse.core.EventCode
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.receiving.Receipt
+import org.pih.warehouse.receiving.ReceiptService
+import org.pih.warehouse.shipping.Shipment
+import org.pih.warehouse.shipping.ShipmentService
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+@Unroll
+class ReceiptServiceSpec extends Specification implements ServiceUnitTest<ReceiptService>, DataTest {
+
+    @Shared
+    ShipmentService shipmentService
+
+    void setupSpec() {
+        mockDomain Receipt
+        mockDomain Shipment
+    }
+
+    void setup() {
+        shipmentService = Mock(ShipmentService) {
+            createShipmentEvent(_, _, _, _) >> void
+        }
+        service.shipmentService = shipmentService
+    }
+
+    void 'savePartialReceiptEvent should create RECEIVED event when partial receiving is not supported'() {
+        given:
+        Location location = Stub(Location) {
+            supports(_ as ActivityCode) >> false
```

Comment:
Is there any reason not to specify the actual enum value here? It makes things a bit more clear in my opinion and also guarantees that we're entering the proper flow.

Same for all the other tests.

```suggestion
            supports(ActivityCode.PARTIAL_RECEIVING) >> false
```